### PR TITLE
Add Sphinx extension for listing all hooks for a specific group (fixes #41)

### LINKED ISSUE #41: Add sphinx extension for listing hooks
Every project which uses gossip, should have its own implementation for listing the gossip hooks it supports.

It would be nice to have a sphinx extension which listing all the hooks of a specific gossip group for documentation.

### PR DESCRIPTION
**Pull request by ayalash**

### CODE REVIEW

File: doc/sphinx_ext.rst [hunk 1]
```diff
@@ -0,0 +1,19 @@
+Sphinx Extension
+================
+Some of Gossip depended packages uses hooks as part of their public API. As such, adding the hooks to the documentation is needed.
```
Comment (vmalloc) [inline doc/sphinx_ext.rst]: Some packages depending on gossip use hooks as part of their public APIs.

Comment (vmalloc) [reply doc/sphinx_ext.rst]: In such cases, adding hook information to their respective documentation pages may be needed.

File: doc/sphinx_ext.rst [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+Gossip has sphinx exntension implementation. For using it:
```
Comment (vmalloc) [inline doc/sphinx_ext.rst]: Gossip supports doing just that through its sphinx extension. In order to use it, install gossip with...

**Comment by ayalash** [comment]
@vmalloc Fixed & Pushed.
I'm not sure that I understood correctly what you wanted with the rst, so please review it after my changes.

**Pull request merged**